broker/bam/inc/com/centreon/broker/bam/availability_builder.hh

```diff
@@ -56,7 +54,8 @@ class availability_builder {
                  time_t start,
                  time_t end,
                  bool was_in_downtime,
-                 time::timeperiod::ptr const& tp);
+                 time::timeperiod::ptr const& tp,
+                 const std::shared_ptr<spdlog::logger>& logger);
```

Comment:
it would be better to store logger in object as for availability_thread

Reply:
I didn't want to do that because the logger is not used except in this function. It is always possible to include the logger in the object when we'll use it more.

Reply:
And it is only used in trace mode... So very rarely.